datafusion/physical-expr/src/intervals/interval_aritmetic.rs

```diff
@@ -1221,21 +1221,29 @@ mod tests {
 
     #[test]
     fn and_test() -> Result<()> {
+        // check that (lhs AND rhs) is equal to result
+        // ((lhs_lower, lhs_upper), (rhs_lower, rhs_upper), lower_result, upper_result))
         let cases = vec![
-            (false, true, false, false, false, false),
-            (false, false, false, true, false, false),
-            (false, true, false, true, false, true),
-            (false, true, true, true, false, true),
-            (false, false, false, false, false, false),
-            (true, true, true, true, true, true),
+            // Note there are only three valid boolean intervals: [false, false], [false, true] and [true, true],
```

Comment:
I added the three missing cases and  reordered the cases so it was easier to see what is covered and what was not.